test/unit/job_invocation_composer_test.rb

```diff
@@ -925,12 +925,14 @@ def provider_inputs
 
     context 'with template in setting present' do
       before do
-        FactoryBot.create(
-          :setting,
-          :name => 'remote_execution_form_job_template',
-          :category => 'Setting::RemoteExecution',
-          :value => setting_template.name
-        )
+        Foreman.settings._add('remote_execution_form_job_template',
+          context: :test,
+          type: :string,
+          category: 'Setting::RemoteExecution',
+          full_name: '',
+          description: '',
+          default: '')
+        Setting[:remote_execution_form_job_template] = setting_template.name
       end
```

Comment:
@ezr-ondrej is this how settings in tests are supposed to be handled?

Reply:
^^